include/mbedtls/x509_crt.h

```diff
@@ -1137,6 +1137,23 @@ int mbedtls_x509write_crt_set_key_usage( mbedtls_x509write_cert *ctx,
 int mbedtls_x509write_crt_set_ns_cert_type( mbedtls_x509write_cert *ctx,
                                     unsigned char ns_cert_type );
 
+/**
+ * \brief               Query certificate for given extension type
+ *
+ * \param[in] ctx       Certificate context to be queried, must not be \c NULL
+ * \param ext_type      Extension type being queried for, must be a valid
+ *                      extension type. Must be one of the MBEDTLS_X509_EXT_XXX
+ *                      values
+ *
+ * \return              0 if the given extension type is not present,
+ *                      non-zero otherwise
+ */
+static inline int mbedtls_x509_crt_has_ext_type( const mbedtls_x509_crt *ctx,
```

Comment:
This is guarded by MBEDTLS_X509_CRT_WRITE_C (as per the test code), but I don't think it should be. Please move outside of the guard (and update the tests to not depend on it), unless there's a reason we want to do this?

Reply:
I think you're right actually. Initially I thought it needed the guard but revisiting this I can't now see a reason. I will move the definition outside and update the tests. Thanks for the review.